obs: Add CentOS 8

GabyCT opened the pull request:
Build packages for CentOS 8.

Fixes #1048

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

GabyCT commented:
/AzurePipelines run

GabyCT commented:
/AzurePipelines run

GabyCT commented:
@jcvenegas , I am getting the following error
```
Project home:katacontainers:ci:x86_64:packaging-PR-1049 has blocked packages, waiting
    <status package="linux-container" code="unresolvable">
    <status package="qemu-vanilla" code="unresolvable">
```
however, I am not sure how to fix this or if this PR has anything to do with this ..thanks

GabyCT commented:
/AzurePipelines run

GabyCT commented:
/AzurePipelines run

jodh-intel commented on obs-packaging/linux-container/kata-linux-container.spec-template:
```diff
@@ -30,6 +30,10 @@ BuildRequires:  libelf-devel
 %endif
 
+%if 0%{?centos_version}
+BuildRequires:  elfutils-libelf-devel
```
Won't this change affect Centos 7 and 8? If so, why didn't we need this for 7 before? Same comment for `python2` below.

GabyCT commented:
/AzurePipelines run

GabyCT commented:
/AzurePipelines run

GabyCT commented:
/AzurePipelines run

GabyCT commented:
/AzurePipelines run

GabyCT commented:
/AzurePipelines run

GabyCT commented:
/AzurePipelines run

GabyCT commented:
/AzurePipelines run

GabyCT commented:
/AzurePipelines run

GabyCT commented:
/AzurePipelines run

GabyCT commented:
/AzurePipelines run

GabyCT commented:
/AzurePipelines run

GabyCT commented:
/AzurePipelines run

GabyCT commented:
/AzurePipelines run

GabyCT commented:
@rhafer, I am trying to build the obs package for CentOS 8, however, I am getting a weird issue https://build.opensuse.org/project/monitor/home:katacontainers:ci:x86_64:packaging-PR-1049?arch_x86_64=1&defaults=0&repo_CentOS_8=1&unresolvable=1, it seems that `python2` and `python2-devel` are not provided, it also happens if I changed it to `python3` and `python3-devel`. How can I add `python`? thanks

rhafer commented:
> @rhafer, I am trying to build the obs package for CentOS 8, however, I am getting a weird issue https://build.opensuse.org/project/monitor/home:katacontainers:ci:x86_64:packaging-PR-1049?arch_x86_64=1&defaults=0&repo_CentOS_8=1&unresolvable=1, it seems that `python2` and `python2-devel` are not provided, it also happens if I changed it to `python3` and `python3-devel`. How can I add `python`? thanks

I think that is because of the "modules" concept in CentOS8. In order to install python3 you need to enable the python36 module. This can be done "Project Config" in the Build Service. See here for some more details: https://lists.opensuse.org/opensuse-buildservice/2019-11/msg00024.html

GabyCT commented:
/AzurePipelines run

GabyCT commented:
@rhafer thanks for the info, I added the `ExpandedFlags` at the project configuration at https://build.opensuse.org/project/show/home:katacontainers:ci:x86_64:packaging-PR-1049 and it worked, however, is it possible to pass that `ExpandFlags: module:python36-3.6` as a file (as a meta)? The issue is that the CentOS 8 packages will pass and will be build because of that flag but only for that PR, in case that someone submitted another PR then this will failed. I tried to put that flag in the project configuration of https://build.opensuse.org/project/show/home:katacontainers:ci and removed the flag from the PR in order to see if the project will inherit the flag and again the build of the packages of CentOS 8 failed, so I think that we need to pass that info as a file, is that possible?

rhafer commented:
> @rhafer thanks for the info, I added the `ExpandedFlags` at the project configuration at https://build.opensuse.org/project/show/home:katacontainers:ci:x86_64:packaging-PR-1049 and it worked, however, is it possible to pass that `ExpandFlags: module:python36-3.6` as a file (as a meta)?

Yes. That should be possible with:

`osc meta prjconf -F <file>`

I guess that could be put into `create-repo-branch.sh`

> I tried to put that flag in the project configuration of https://build.opensuse.org/project/show/home:katacontainers:ci and removed the flag from the PR in order to see if the project will inherit the flag

Hm, I think the project configuration will only be inherited when you have the project created as part of the branch command. I think this doesn't currently fit the existing workflow. So uploading the config with the above command should be fine.

GabyCT commented:
/AzurePipelines run

GabyCT commented:
/AzurePipelines run

GabyCT commented:
/AzurePipelines run

GabyCT commented:
/AzurePipelines run

GabyCT commented:
/AzurePipelines run

GabyCT commented:
/AzurePipelines run

GabyCT commented:
/test

GabyCT commented:
@jodh-intel changes applied

GabyCT commented:
@rhafer thank you so much your solution worked :)
